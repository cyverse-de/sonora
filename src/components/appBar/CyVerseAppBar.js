/**
 *
 * @author sriram
 * A component that displays app level tool bar with search and user menu.
 *
 */

import React from "react";
import intlData from "./messages";
import ids from "./ids";
import constants from "../../constants";

import { build, formatMessage, withI18N } from "@cyverse-de/ui-lib";

import { useRouter } from "next/router";
import { injectIntl } from "react-intl";
import {
    AppBar,
    Badge,
    Button,
    ButtonGroup,
    ClickAwayListener,
    Grow,
    IconButton,
    InputBase,
    MenuItem,
    MenuList,
    Paper,
    Popper,
    Toolbar,
} from "@material-ui/core";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";
import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import NotificationsIcon from "@material-ui/icons/Notifications";
import AccountCircle from "@material-ui/icons/AccountCircle";

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
    appBar: {
        backgroundColor: theme.palette.lightGray,
        boxShadow: 0,
    },

    search: {
        position: "relative",
        borderRadius: theme.shape.borderRadius,
        backgroundColor: theme.palette.white,
        "&:hover": {
            backgroundColor: theme.palette.white,
        },
        marginRight: 0,
        marginLeft: 0,
        width: "100%",
        [theme.breakpoints.up("sm")]: {
            marginLeft: theme.spacing(2),
            width: "100%",
        },
    },
    searchIcon: {
        width: theme.spacing(7),
        height: "100%",
        position: "absolute",
        pointerEvents: "none",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
    },

    sectionDesktop: {
        display: "none",
        [theme.breakpoints.up("md")]: {
            display: "flex",
        },
    },
    inputInput: {
        padding: theme.spacing(1, 1, 1, 7),
        transition: theme.transitions.create("width"),
        width: "100%",
        [theme.breakpoints.up("sm")]: {
            width: 120,
            "&:focus": {
                width: 200,
            },
        },
    },
}));
const searchOptions = ["All", "Data", "Apps", "Analyses"];

function CyverseAppBar(props) {
    const theme = useTheme();
    const classes = useStyles();
    const router = useRouter();
    const { intl, children } = props;

    const handleUserButtonClick = (event) => {
        const { user } = props.children.props;
        if (!user) {
            router.push("/login");
        }
    };

    return (
        <React.Fragment>
            <div className={classes.root}>
                <AppBar
                    position="static"
                    variant="outlined"
                    className={classes.appBar}
                >
                    <Toolbar>
                        <a
                            href={constants.CYVERSE_URL}
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            <img src="/de.png" alt="CyVerse"></img>
                        </a>
                        <div className={classes.search}>
                            <div className={classes.searchIcon}>
                                <SearchIcon color="primary" />
                            </div>
                            <InputBase
                                placeholder="Searchâ€¦"
                                classes={{
                                    root: classes.inputRoot,
                                    input: classes.inputInput,
                                }}
                                inputProps={{ "aria-label": "search" }}
                            />
                        </div>
                        <div>
                            <SearchOptions intl={intl} />
                        </div>

                        <div className={classes.root} />
                        <div style={{ display: "flex" }}>
                            <IconButton
                                aria-label="show new notifications"
                                color="primary"
                            >
                                <Badge badgeContent={17} color="error">
                                    <NotificationsIcon />
                                </Badge>
                            </IconButton>
                            <IconButton
                                edge="end"
                                aria-label="account of current user"
                                aria-controls="menudId1"
                                aria-haspopup="true"
                                color="primary"
                                onClick={handleUserButtonClick}
                            >
                                <AccountCircle />
                            </IconButton>
                        </div>
                    </Toolbar>
                </AppBar>
            </div>
            {children}
        </React.Fragment>
    );
}

function SearchOptions(props) {
    const { intl } = props;
    const theme = useTheme();
    const [open, setOpen] = React.useState(false);
    const anchorRef = React.useRef(null);
    const [selectedIndex, setSelectedIndex] = React.useState(0);

    const handleClick = () => {
        //as of now do nothing
        console.info(`You clicked ${searchOptions[selectedIndex]}`);
    };

    const handleMenuItemClick = (event, index) => {
        setSelectedIndex(index);
        setOpen(false);
    };

    const handleToggle = () => {
        setOpen((prevOpen) => !prevOpen);
    };

    const handleClose = (event) => {
        if (anchorRef.current && anchorRef.current.contains(event.target)) {
            return;
        }

        setOpen(false);
    };
    return (
        <Paper>
            <ButtonGroup
                variant="contained"
                ref={anchorRef}
                aria-label={formatMessage(intl, "searchFilterAriaLabel")}
            >
                <Button
                    id={build(ids.APP_BAR_BASE_ID, ids.SEARCH_FILTER_BTN)}
                    onClick={handleClick}
                    style={{
                        backgroundColor: theme.palette.primary.main,
                        color: theme.palette.white,
                    }}
                >
                    {searchOptions[selectedIndex]}
                </Button>
                <Button
                    id={build(
                        ids.APP_BAR_BASE_ID,
                        ids.SEARCH_FILTER_OPTIONS_BTN
                    )}
                    style={{
                        backgroundColor: theme.palette.primary.main,
                        color: theme.palette.white,
                    }}
                    size="small"
                    aria-controls={
                        open
                            ? formatMessage(intl, "searchFilterAriaControl")
                            : undefined
                    }
                    aria-expanded={open ? "true" : undefined}
                    aria-label={formatMessage(
                        intl,
                        "searchFilterMenuAriaLabel"
                    )}
                    aria-haspopup="menu"
                    onClick={handleToggle}
                >
                    <ArrowDropDownIcon />
                </Button>
            </ButtonGroup>
            <Popper
                open={open}
                anchorEl={anchorRef.current}
                role={undefined}
                transition
                disablePortal
            >
                {({ TransitionProps, placement }) => (
                    <Grow
                        {...TransitionProps}
                        style={{
                            transformOrigin:
                                placement === "bottom"
                                    ? "center top"
                                    : "center bottom",
                        }}
                    >
                        <Paper>
                            <ClickAwayListener onClickAway={handleClose}>
                                <MenuList
                                    id={build(
                                        ids.APP_BAR_BASE_ID,
                                        ids.SEARCH_FILTER_MENU
                                    )}
                                >
                                    {searchOptions.map((option, index) => (
                                        <MenuItem
                                            key={option}
                                            id={build(
                                                ids.APP_BAR_BASE_ID,
                                                ids.SEARCH_FILTER_MENU +
                                                    "." +
                                                    option
                                            )}
                                            selected={index === selectedIndex}
                                            onClick={(event) =>
                                                handleMenuItemClick(
                                                    event,
                                                    index
                                                )
                                            }
                                        >
                                            {option}
                                        </MenuItem>
                                    ))}
                                </MenuList>
                            </ClickAwayListener>
                        </Paper>
                    </Grow>
                )}
            </Popper>
        </Paper>
    );
}

export default withI18N(injectIntl(CyverseAppBar), intlData);
