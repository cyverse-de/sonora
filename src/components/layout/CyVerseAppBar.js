/**
 *
 * @author sriram
 * A component that displays app level tool bar with search and user menu. Also renders a drawer menu
 * in small screens instead of Navigation bar
 *
 */

import React, { useEffect, useRef, useState } from "react";
import clsx from "clsx";
import { useQuery } from "react-query";
import { useRouter } from "next/router";
import { injectIntl } from "react-intl";

import ids from "./ids";
import intlData from "./messages";
import constants from "../../constants";
import GlobalSearchField from "../search/GlobalSearchField";
import NavigationConstants from "../../common/NavigationConstants";
import Notifications from "./Notifications";
import CustomIntercom from "./CustomIntercom";
import { useUserProfile } from "../../contexts/userProfile";
import { getUserProfile } from "../../serviceFacades/users";

import {
    build,
    CyVerseAnnouncer,
    formatHTMLMessage,
    formatMessage,
    getMessage,
    withI18N,
} from "@cyverse-de/ui-lib";

import {
    AppBar,
    Avatar,
    Divider,
    Drawer,
    Hidden,
    IconButton,
    List,
    ListItem,
    ListItemIcon,
    ListItemText,
    Toolbar,
    Tooltip,
    Typography,
    useTheme,
} from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";
import AccountCircle from "@material-ui/icons/AccountCircle";
import MenuIcon from "@material-ui/icons/Menu";
import SettingsIcon from "@material-ui/icons/Settings";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import ChevronRightIcon from "@material-ui/icons/ChevronRight";
import SupervisorAccountIcon from "@material-ui/icons/SupervisorAccount";
import LabelImportantIcon from "@material-ui/icons/LabelImportant";

const drawerWidth = 200;
const miniDrawerWidth = 65;
const useStyles = makeStyles((theme) => ({
    root: {
        flex: 1,
    },
    appBar: {
        backgroundColor: theme.palette.primary,
        zIndex: theme.zIndex.drawer + 1,
        width: "100%",
        [theme.breakpoints.up("sm")]: {
            marginLeft: miniDrawerWidth,
            transition: theme.transitions.create(["width", "margin"], {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen,
            }),
            width: `calc(100% - ${miniDrawerWidth + 1}px)`,
        },
    },
    appBarShift: {
        [theme.breakpoints.up("sm")]: {
            marginLeft: drawerWidth,
            width: `calc(100% - ${drawerWidth}px)`,
            transition: theme.transitions.create(["width", "margin"], {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen,
            }),
        },
    },
    drawerIcon: {
        flexShrink: 0,
        marginRight: theme.spacing(3.5),
        width: 32,
    },
    userIcon: {
        cursor: "pointer",
        backgroundColor: theme.palette.success.main,
        color: theme.palette.success.contrastText,
        width: theme.spacing(5),
        height: theme.spacing(5),
        marginLeft: theme.spacing(1),
        "&:hover": {
            textDecoration: "underline",
            textDecorationColor: theme.palette.success.contrastText,
        },
    },
    accountIcon: {
        cursor: "pointer",
        color: theme.palette.primary.contrastText,
        backgroundColor: theme.palette.primary.main,
        width: theme.spacing(5),
        height: theme.spacing(5),
    },
    icon: {
        color: theme.palette.info.contrastText,
    },
    appBarIcon: {
        [theme.breakpoints.up("sm")]: {
            margin: theme.spacing(2),
        },
        [theme.breakpoints.down("xs")]: {
            margin: theme.spacing(0.2),
        },
    },
    drawer: {
        width: drawerWidth,
        flexShrink: 0,
        whiteSpace: "nowrap",
    },
    drawerOpen: {
        backgroundColor: theme.palette.info.main,
        color: theme.palette.info.contrastText,
        width: drawerWidth,
        [theme.breakpoints.up("sm")]: {
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen,
            }),
        },
    },
    drawerClose: {
        [theme.breakpoints.up("sm")]: {
            backgroundColor: theme.palette.info.main,
            color: theme.palette.info.contrastText,
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen,
            }),
            overflowX: "hidden",
            width: miniDrawerWidth,
        },
    },
    content: {
        width: "100%",
        [theme.breakpoints.up("sm")]: {
            width: `calc(100% - ${miniDrawerWidth + 1}px)`,
            marginLeft: miniDrawerWidth,
            padding: theme.spacing(1),
        },
    },
    contentShift: {
        [theme.breakpoints.up("sm")]: {
            marginLeft: drawerWidth,
            width: `calc(100% - ${drawerWidth}px)`,
            transition: theme.transitions.create(["width", "margin"], {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen,
            }),
        },
    },
    profile: {
        cursor: "pointer",
        "&:hover": {
            textDecoration: "underline",
            textDecorationColor: theme.palette.info.contrastText,
        },
    },
    listItem: {
        cursor: "pointer",
        "&:hover": {
            textDecoration: "underline",
            backgroundColor: theme.palette.primary.main,
        },
        color: theme.palette.info.contrastText,
    },
    listItemActive: {
        cursor: "pointer",
        textDecoration: "underline",
        color: theme.palette.primary.contrastText,
        backgroundColor: theme.palette.primary.main,
    },
    menuIcon: {
        color: theme.palette.primary.contrastText,
        [theme.breakpoints.up("sm")]: {
            margin: theme.spacing(0.5),
            color: theme.palette.info.contrastText,
        },
    },
    nested: {
        paddingLeft: theme.spacing(4),
    },
}));

function CyverseAppBar(props) {
    const classes = useStyles();
    const theme = useTheme();
    const router = useRouter();
    const ref = useRef();

    const { intl, children, activeView, setAppBarRef } = props;
    const [userProfile, setUserProfile] = useUserProfile();
    const [avatarLetter, setAvatarLetter] = useState("");
    const [open, setOpen] = useState(false);
    const [adminUser, setAdminUser] = useState(false);

    useQuery({
        queryKey: "getUserProfile",
        queryFn: getUserProfile,
        config: {
            onSuccess: setUserProfile,
        },
    });

    useEffect(() => {
        setAppBarRef(ref);
    }, [ref, setAppBarRef]);

    React.useEffect(() => {
        if (userProfile?.id) {
            setAvatarLetter(userProfile.id.charAt(0).toUpperCase());
            const adminGroups = process.env.ADMIN_GROUPS;
            const groupAttribute = process.env.ADMIN_GROUP_ATTRIBUTE;
            const userGroupMemberships = userProfile.attributes[groupAttribute];
            const adminMemberships = [];
            if (adminGroups && userGroupMemberships) {
                const adminGroupArray = adminGroups.split(",");
                adminGroupArray.forEach((adminGroup) => {
                    const found = userGroupMemberships.find(
                        (userGroup) => adminGroup === userGroup
                    );
                    if (found) {
                        adminMemberships.push(found);
                    }
                });
                setAdminUser(adminMemberships.length > 0);
            }
            console.log("is admin user => " + adminUser);
        }
    }, [userProfile, adminUser, setAdminUser, setAvatarLetter]);

    const handleUserButtonClick = (event) => {
        toggleDrawer(false);
        if (!userProfile) {
            router.push(`/${NavigationConstants.LOGIN}${router.asPath}`);
        } else {
            router.push(`/${NavigationConstants.LOGOUT}`);
        }
    };
    const handleSearchClick = (event) => {
        router.push("/" + NavigationConstants.SEARCH);
        toggleDrawer(false);
    };
    const handleDrawerOpen = () => {
        setOpen(true);
    };

    const handleDrawerClose = () => {
        setOpen(false);
    };

    const toggleDrawer = (open) => (event) => {
        setOpen(open);
    };

    const accountAvatar = (
        <Avatar
            className={userProfile ? classes.userIcon : classes.accountIcon}
            onClick={handleUserButtonClick}
        >
            {userProfile ? (
                <Tooltip
                    title={getMessage("logout", {
                        values: { name: userProfile.id },
                    })}
                    aria-label={formatMessage(intl, "logout", {
                        name: userProfile.id,
                    })}
                    placement="bottom"
                    arrow
                >
                    <Typography variant={"h6"}>{avatarLetter}</Typography>
                </Tooltip>
            ) : (
                <Tooltip
                    title={getMessage("login")}
                    aria-label={formatMessage(intl, "login")}
                    placement="bottom"
                    arrow
                >
                    <AccountCircle fontSize="large" />
                </Tooltip>
            )}
        </Avatar>
    );

    const drawerItems = (
        <List>
            <Tooltip
                title={formatMessage(intl, "dashboard")}
                placement="right"
                arrow
            >
                <ListItem
                    id={build(ids.DRAWER_MENU, ids.DASHBOARD_MI)}
                    onClick={() => {
                        toggleDrawer(false);
                        router.push("/" + NavigationConstants.DASHBOARD);
                    }}
                    className={
                        activeView === NavigationConstants.DASHBOARD
                            ? classes.listItemActive
                            : classes.listItem
                    }
                >
                    <img
                        className={classes.drawerIcon}
                        src="/dashboard_selected.png"
                        alt={formatMessage(intl, "dashboard")}
                    />
                    <ListItemText>{getMessage("dashboard")}</ListItemText>
                </ListItem>
            </Tooltip>
            <Tooltip
                title={formatMessage(intl, "data")}
                placement="right"
                arrow
            >
                <ListItem
                    id={build(ids.DRAWER_MENU, ids.DATA_MI)}
                    onClick={() => {
                        toggleDrawer(false);
                        router.push("/" + NavigationConstants.DATA);
                    }}
                    className={
                        activeView === NavigationConstants.DATA
                            ? classes.listItemActive
                            : classes.listItem
                    }
                >
                    <img
                        className={classes.drawerIcon}
                        src="/data_selected.png"
                        alt={formatMessage(intl, "data")}
                    />
                    <ListItemText>{getMessage("data")}</ListItemText>
                </ListItem>
            </Tooltip>
            <Tooltip
                title={formatMessage(intl, "apps")}
                placement="right"
                arrow
            >
                <ListItem
                    id={build(ids.DRAWER_MENU, ids.APPS_MI)}
                    onClick={() => {
                        toggleDrawer(false);
                        router.push("/" + NavigationConstants.APPS);
                    }}
                    className={
                        activeView === NavigationConstants.APPS
                            ? classes.listItemActive
                            : classes.listItem
                    }
                >
                    <img
                        className={classes.drawerIcon}
                        src="/apps_selected.png"
                        alt={formatMessage(intl, "apps")}
                    />
                    <ListItemText>{getMessage("apps")}</ListItemText>
                </ListItem>
            </Tooltip>
            <Tooltip
                title={formatMessage(intl, "analyses")}
                placement="right"
                arrow
            >
                <ListItem
                    id={build(ids.DRAWER_MENU, ids.ANALYSES_MI)}
                    onClick={() => {
                        toggleDrawer(false);
                        router.push("/" + NavigationConstants.ANALYSES);
                    }}
                    className={
                        activeView === NavigationConstants.ANALYSES
                            ? classes.listItemActive
                            : classes.listItem
                    }
                >
                    <img
                        className={classes.drawerIcon}
                        src="/analyses_selected.png"
                        alt={formatMessage(intl, "analyses")}
                    />
                    <ListItemText>{getMessage("analyses")}</ListItemText>
                </ListItem>
            </Tooltip>
            <Hidden only={["md", "lg", "xl"]}>
                <Tooltip
                    title={formatMessage(intl, "search")}
                    placement="right"
                    arrow
                >
                    <ListItem
                        id={build(ids.DRAWER_MENU, ids.SEARCH_MI)}
                        onClick={handleSearchClick}
                        className={
                            activeView === NavigationConstants.SEARCH
                                ? classes.listItemActive
                                : classes.listItem
                        }
                    >
                        <ListItemIcon>
                            <SearchIcon
                                className={classes.icon}
                                fontSize="large"
                            />
                        </ListItemIcon>
                        <ListItemText>{getMessage("search")}</ListItemText>
                    </ListItem>
                </Tooltip>
            </Hidden>
            <Divider />
            <Tooltip
                title={formatMessage(intl, "settings")}
                placement="right"
                arrow
            >
                <ListItem
                    id={build(ids.DRAWER_MENU, ids.SETTINGS_MI)}
                    onClick={() =>
                        router.push("/" + NavigationConstants.SETTINGS)
                    }
                    className={
                        activeView === NavigationConstants.SETTINGS
                            ? classes.listItemActive
                            : classes.listItem
                    }
                >
                    <ListItemIcon>
                        <SettingsIcon
                            className={classes.icon}
                            fontSize="large"
                        />
                    </ListItemIcon>
                    <ListItemText>{getMessage("settings")}</ListItemText>
                </ListItem>
            </Tooltip>
        </List>
    );

    const adminDrawerItems = (
        <List>
            <ListItem
                id={build(ids.DRAWER_MENU, ids.ADMIN_MI)}
                className={classes.listItem}
            >
                <ListItemIcon>
                    <SupervisorAccountIcon
                        className={classes.icon}
                        fontSize="large"
                    />
                </ListItemIcon>
                <ListItemText>{getMessage("admin")}</ListItemText>
            </ListItem>
            <List component="div" disablePadding>
                <Tooltip
                    title={formatMessage(intl, "vice")}
                    placement="right"
                    arrow
                >
                    <ListItem
                        button
                        id={build(ids.DRAWER_MENU, ids.VICE_MI)}
                        className={clsx(
                            classes.nested,
                            activeView === NavigationConstants.VICE
                                ? classes.listItemActive
                                : classes.listItem
                        )}
                        onClick={() =>
                            router.push(
                                "/" +
                                    NavigationConstants.ADMIN +
                                    "/" +
                                    NavigationConstants.VICE
                            )
                        }
                    >
                        <ListItemIcon>
                            <LabelImportantIcon className={classes.icon} />
                        </ListItemIcon>
                        <ListItemText>{getMessage("vice")}</ListItemText>
                    </ListItem>
                </Tooltip>
            </List>
        </List>
    );
    return (
        <div className={classes.root}>
            <AppBar
                id={ids.APP_BAR_BASE}
                position="static"
                variant="outlined"
                ref={ref}
                className={clsx(classes.appBar, {
                    [classes.appBarShift]: open,
                })}
            >
                <Toolbar>
                    <Hidden only={["sm", "md", "lg", "xl"]}>
                        <IconButton
                            aria-label={formatMessage(intl, "openDrawer")}
                            onClick={handleDrawerOpen}
                            edge="start"
                            className={classes.menuIcon}
                        >
                            <MenuIcon />
                        </IconButton>
                        <Typography>
                            {formatHTMLMessage("discovery")}
                            &nbsp;{formatHTMLMessage("environment")}
                        </Typography>
                    </Hidden>
                    <Hidden xsDown>
                        <a
                            href={constants.CYVERSE_URL}
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            <img
                                src="/de_white.png"
                                alt={formatMessage(intl, "cyverse")}
                            ></img>
                        </a>
                    </Hidden>
                    <Hidden smDown>
                        <GlobalSearchField />
                    </Hidden>
                    <div className={classes.root} />
                    <div style={{ display: "flex" }}>
                        <CustomIntercom intl={intl} />
                        <Notifications intl={intl} />
                    </div>
                    <Hidden only={["xs"]}>
                        <div id={build(ids.DRAWER_MENU, ids.ACCOUNT_MI)}>
                            {accountAvatar}
                        </div>
                    </Hidden>
                </Toolbar>
            </AppBar>
            <Hidden xsDown>
                <Drawer
                    variant="permanent"
                    className={clsx(classes.drawer, {
                        [classes.drawerOpen]: open,
                        [classes.drawerClose]: !open,
                    })}
                    classes={{
                        paper: clsx({
                            [classes.drawerOpen]: open,
                            [classes.drawerClose]: !open,
                        }),
                    }}
                >
                    <div className={classes.toolbar}>
                        {!open && (
                            <IconButton
                                aria-label="open drawer"
                                onClick={handleDrawerOpen}
                                edge="start"
                                className={clsx(classes.menuIcon, {
                                    [classes.hide]: open,
                                })}
                            >
                                <MenuIcon fontSize="large" />
                            </IconButton>
                        )}
                        {open && (
                            <IconButton
                                onClick={handleDrawerClose}
                                className={classes.menuIcon}
                            >
                                {theme.direction === "rtl" ? (
                                    <ChevronRightIcon fontSize="large" />
                                ) : (
                                    <ChevronLeftIcon fontSize="large" />
                                )}
                            </IconButton>
                        )}
                    </div>
                    <Divider />
                    {drawerItems}
                    {open && adminUser && (
                        <>
                            <Divider />
                            {adminDrawerItems}
                        </>
                    )}
                </Drawer>
            </Hidden>
            <Hidden only={["sm", "md", "lg", "xl"]}>
                <Drawer
                    className={clsx(classes.drawer, {
                        [classes.drawerOpen]: open,
                        [classes.drawerClose]: !open,
                    })}
                    classes={{
                        paper: clsx({
                            [classes.drawerOpen]: open,
                            [classes.drawerClose]: !open,
                        }),
                    }}
                    open={open}
                    onClose={toggleDrawer(false)}
                >
                    <div
                        id={build(ids.DRAWER_MENU, ids.ACCOUNT_MI)}
                        style={{ margin: 8 }}
                    >
                        {accountAvatar}
                    </div>
                    <Divider />
                    {drawerItems}
                    {adminUser && (
                        <>
                            <Divider />
                            {adminDrawerItems}
                        </>
                    )}
                </Drawer>
            </Hidden>
            <CyVerseAnnouncer />
            <main
                className={clsx(classes.content, {
                    [classes.contentShift]: open,
                })}
            >
                {children}
            </main>
        </div>
    );
}

export default withI18N(injectIntl(CyverseAppBar), intlData);
