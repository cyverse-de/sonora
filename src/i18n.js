const NextI18Next = require("next-i18next").default;

module.exports = new NextI18Next({
    defaultLanguage: "en",
});
