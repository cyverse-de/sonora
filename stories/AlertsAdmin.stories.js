import React from "react";
import AlertsEditor from "../src/components/alerts/admin/AlertsAdmin";
import { mockAxios } from "./axiosMock";

//import testConfig from "./configMock";

const emptyAlerts = { alerts: [] };
const nonEmptyAlerts = {
    alerts: [
        {
            end_date: "2024-01-01T00:00:00Z",
            alert: "This is an alert in the past.",
        },
        {
            end_date: "2100-01-01T00:00:00Z",
            alert: "This is a testing alert far in the future.",
        },
    ],
};

const alertsAdminTestTemplate = (args) => {
    const { mockAllAlerts } = args;
    //mockAxios.onGet("/api/...").reply(200, mockDataSpec)
    mockAxios.onGet("/api/alerts/all").reply(200, mockAllAlerts);
    mockAxios.onPost("/api/admin/alerts").reply((config) => {
        const {
            "alert-text": alert,
            "start-date": start_date,
            "end-date": end_date,
        } = JSON.parse(config.data);
        mockAllAlerts.alerts.push({ alert, start_date, end_date });
        return [201, {}];
    });
    mockAxios.onDelete("/api/admin/alerts").reply((config) => {
        const d = JSON.parse(config.data);
        const deleteIndex = mockAllAlerts.alerts.findIndex(
            (alert) =>
                alert.alert === d["alert-text"] &&
                alert.end_date === d["end-date"]
        );
        if (deleteIndex !== -1) {
            mockAllAlerts.alerts.splice(deleteIndex, 1);
        }
        return deleteIndex !== -1 ? [200, {}] : [500, {}];
    });
    return <AlertsEditor />;
};

export const NormalView = alertsAdminTestTemplate.bind({});
NormalView.args = {
    mockAllAlerts: emptyAlerts,
};

export const HasAlertsView = alertsAdminTestTemplate.bind({});
HasAlertsView.args = {
    mockAllAlerts: nonEmptyAlerts,
};

export default {
    title: "Alerts / Alerts Editor",
    component: AlertsEditor,
    parameters: {
        // This is the max delay allowed.
        chromatic: { delay: 15000 },
    },
};
