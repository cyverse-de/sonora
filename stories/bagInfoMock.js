const bagInfoMock = {
    id: "d2b9e454-0038-11ec-860b-008cfa5ae621",
    user_id: "6be9b792-854a-11e4-b877-cb0cf45dbbb0",

    contents: {
        items: [
            {
                infoType: "ht-analysis-path-list",
                path: "/iplant/home/test/5r43",
                fileSize: 72,
                permission: "own",
                type: "file",
                badName: false,
                dateModified: 1607364685000,
                isFavorite: false,
                label: "5r43",
                id: "9eb49166-38b7-11eb-93a1-90e2ba675364",
                dateCreated: 1607364685000,
            },
        ],
    },
};

export default bagInfoMock;
