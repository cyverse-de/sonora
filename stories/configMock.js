import constants from "../src/constants";

export default {
    intercom: {
        appId: "appId",
        enabled: true,
        companyId: "companyId",
        companyName: "companyName",
    },
    sessions: {
        poll_interval_ms: 5000,
    },
    admin: {
        groups: "test",
        group_attribute_name: "entitlement",
    },
    analysis: { supportUser: { user: "support_user", source_id: "ldap" } },
    irods: {
        home_path: "/iplant/home",
        trash_path: "/iplant/trash/home/de-irods",
    },
    tools: {
        admin: {
            max_cpu_limit: 48,
            max_memory_limit: 244 * constants.ONE_GiB,
            max_disk_limit: 1024 * constants.ONE_GiB,
        },
        private: {
            max_cpu_limit: 8,
            max_memory_limit: 16 * constants.ONE_GiB,
            max_disk_limit: 512 * constants.ONE_GiB,
        },
    },
    vice: {
        initContainerName: "input-files-init",
        inputFilesContainerName: "input-files",
        viceProxyContainerName: "vice-proxy",
        analysisContainerName: "analysis",
        deploymentTimeoutMs: 15000,
        defaultImage: "discoenv/cas-proxy",
        defaultName: "cas-proxy",
        defaultCasUrl: "https://olson.cyverse.org/cas",
        defaultCasValidate: "validate",
    },
    subscriptions: {
        checkout_url:
            "https://cyverse-subscription-sandbox.phoenixbioinformatics.org",
        enforce: true,
    },
    elasticEnabled: true,
};
