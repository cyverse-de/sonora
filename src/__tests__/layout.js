import React, { Suspense } from "react";
import renderer from "react-test-renderer";
import { I18nextProvider } from "react-i18next";
import { AppBar } from "../../stories/CyVerseAppBar.stories";
import i18n from "../test_i18n";

test("App Bar renders", () => {
    const component = renderer.create(
        <I18nextProvider i18n={i18n}>
            <Suspense fallback="Loading...">
                <AppBar />
            </Suspense>
        </I18nextProvider>
    );
    component.unmount();
});
