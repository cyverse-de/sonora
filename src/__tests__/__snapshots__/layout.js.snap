// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Bar renders 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-2 MuiPaper-outlined"
    id="appBar"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="makeStyles-root-1"
      />
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <button
          aria-controls="chat support button"
          aria-label="chat with CyVerse support"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-appBarIcon-7 MuiIconButton-colorPrimary MuiIconButton-sizeSmall"
          disabled={false}
          id="cyverse_intercom_support"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 2H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h4l3 3 3-3h4c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-6 16h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 11.9 13 12.5 13 14h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                />
              </svg>
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorError MuiBadge-invisible"
              />
            </span>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="show new notifications"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-appBarIcon-7 MuiIconButton-colorPrimary MuiIconButton-sizeSmall"
          disabled={false}
          id="appBar.notificationButton"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorError MuiBadge-invisible"
              >
                0
              </span>
            </span>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <main
    className="makeStyles-content-11"
  />
</div>
`;
